filebeat/input/v2/input-cursor/input.go

```diff
@@ -109,12 +109,16 @@ func (inp *managedInput) Run(
 	defer cancel()
 	ctx.Cancelation = cancelCtx
 
+	// This context isn't directly used, so unregister any metric it might have
+	// registered.
+	ctx.UnregisterMetrics()
```

Comment:
This should probably be removed, the pipeline is not going to add any undesired metrics to a context before passing it to `Run` and we're better off documenting that than adding no-ops just in case.